src/features/navigator/NarrativeControl/Copy.tsx

```diff
@@ -21,23 +24,50 @@ export interface CopyProps extends ControlProps {
 }
 
 export const Copy: FC<CopyProps> = ({ narrativeDoc, modalClose, version }) => {
+  /* hooks */
   const dispatch = useAppDispatch();
   const { formState, getValues, register } = useForm<CopyValues>({
     defaultValues: {
       narrativeCopyName: `${narrativeDoc.narrative_title} - Copy`,
     },
     mode: 'all',
   });
+  const [copyTrigger] = copyNarrative.useMutation();
+  /* derived values */
   const inputRegister = inputRegisterFactory<CopyValues>({
     formState,
     register,
   });
+  const { access_group: wsId, obj_id: objId } = narrativeDoc;
+  const errors = formState.errors;
+  const errorEntries = Object.entries(errors);
+  const formInvalid = errorEntries.length > 0;
+  /* copy narrative callback */
   const copyNarrativeHandler = async () => {
     const { narrativeCopyName: name } = getValues();
-    await TODOAddLoadingState();
-    dispatch(copyNarrative({ wsId: narrativeDoc.access_group, name, version }));
+    const message = `Copy ${wsId}/${objId}/${version} as ${name}.`;
     modalClose();
+    dispatch(copyAction({ wsId: narrativeDoc.access_group, name, version }));
+    try {
+      await copyTrigger({
+        nameNew: name,
+        workspaceRef: `${wsId}/${objId}/${version}`,
+        workspaceId: wsId,
+      }).unwrap();
+      dispatch(setLoading(false));
+    } catch (err) {
+      if (!isKBaseBaseQueryError(err)) {
+        console.error({ err }); // eslint-disable-line no-console
+        toast(ErrorMessage({ err }));
+        return;
+      }
+      toast(ErrorMessage({ err: parseError(err) }));
+      dispatch(setLoading(false));
+      return;
```

Comment:
This seems like something we may do a lot, perhaps worth pulling out into a util, I'm not sure. What do you think?

Reply:
To do that we would need to pass `dispatch` as a parameter, which is somewhat standard, but still a little less readable to developers not used to a more functional style. We'll keep an eye on it.